.github/workflows/run_tests.yml

```diff
@@ -24,134 +24,49 @@ jobs:
-            blueprint_verifiers_placeholder_expression_evaluation_component_test,
-            blueprint_verifiers_placeholder_final_polynomial_check_test,
-            blueprint_verifiers_flexible_swap_test,
-            blueprint_verifiers_flexible_additions_test,
-            blueprint_verifiers_flexible_multiplications_test,
-            blueprint_verifiers_flexible_poseidon_test,
-            blueprint_verifiers_flexible_constant_pow_test,
-            blueprint_verifiers_placeholder_verifier_test
-          ]  # Tests to execute
-        include: # Abused to enable proof generation for some tests; add more as needed
-          - target: blueprint_algebra_fields_plonk_non_native_logic_ops_test
-            add-args: "-DBLUEPRINT_PLACEHOLDER_PROOF_GEN=True"
-          - target: blueprint_algebra_fields_plonk_non_native_lookup_logic_ops_test
-            add-args: "-DBLUEPRINT_PLACEHOLDER_PROOF_GEN=True"
     steps:
-      - uses: cachix/install-nix-action@v23
-      - uses: DeterminateSystems/magic-nix-cache-action@main
-      - uses: DeterminateSystems/nix-installer-action@main
+      # https://github.com/actions/checkout/issues/1552
+      - name: Clean up after previous checkout
+        run: chmod +w -R ${GITHUB_WORKSPACE}; rm -rf ${GITHUB_WORKSPACE}/*;
 
-      - name: Checkout suite
-        uses: actions/checkout@v3
+      - name: Checkout Blueprint
+        uses: actions/checkout@v4
         with:
-          repository: ${{ env.SUITE_REPO }}
-          submodules: recursive
           fetch-depth: 0
+          submodules: recursive
 
       - name: Checkout submodules to specified refs
-        if: needs.handle-syncwith.outputs.prs-refs != ''
-        uses: NilFoundation/ci-cd/actions/recursive-checkout@v1
+        if: inputs.submodules-refs != ''
+        uses: NilFoundation/ci-cd/actions/recursive-checkout@v1.2.1
         with:
+          refs: ${{ inputs.submodules-refs }}
           paths: |
             ${{ github.workspace }}/**
+            !${{ github.workspace }}/
             !${{ github.workspace }}/**/.git/**
-          refs: ${{ needs.handle-syncwith.outputs.prs-refs }}
-
-      - name: Checkout source code
-        uses: actions/checkout@v3
-        with:
-          path: ./libs/${{ env.LIB_NAME }}
-          submodules: recursive
-          fetch-depth: 0
-
-      - name: Configure Nix
-        run: |
-          mkdir -p ~/.config/nix
-          echo "extra-experimental-features = nix-command flakes" > ~/.config/nix/nix.conf
 
-      - name: Cmake and build
+      # nix is taken from the cloud-init template, no need to install it.
+      - name: Build and run tests
         env:
-          CMAKE_ARGS: "
-            -DCMAKE_CXX_STANDARD=17
-            -DCMAKE_BUILD_TYPE=Debug
-            -DBUILD_SHARED_LIBS=FALSE
-            -DENABLE_TESTS=TRUE
-            -DCMAKE_C_COMPILER=clang
-            -DCMAKE_CXX_COMPILER=clang++
-            ${{ matrix.add-args || ''}}"
+          NIX_CONFIG: |
+            cores = 2

```

Comment:
This is kinda strange. Are we trying to lower the load on our runners here or what?

Reply:
Yes, it was failing with OOM otherwise in crypto3